lib/TokenManager.ts

```diff
@@ -393,8 +494,32 @@ export class TokenManager {
         this.remove(key);
       }
     };
-    this.on('expired', onTokenExpiredHandler);
+    this.on(EVENT_EXPIRED, onTokenExpiredHandler);
 
     setExpireEventTimeoutAll(sdk, tokenMgmtRef, storage);
+
+    // Sync authState cross multiple tabs when localStorage is used as the storageProvider
+    window.addEventListener('storage', ({ key, newValue, oldValue }: StorageEvent) => {
+      const emitEventsAndResetExpireEventTimeouts = () => {
+        this._resetExpireEventTimeoutAll();
+        this._emitEventsForCrossTabsStorageUpdate(newValue, oldValue);
+      };
+
+      // Skip if:
+      // not from localStorage.clear (event.key is null)
+      // event.key is not the storageKey
+      // oldValue === newValue
+      if ((key && key !== options.storageKey) || newValue === oldValue) {
+        return;
+      }
+
+      // LocalStorage cross tabs update is not synced in IE, set a 1s timer to read latest value
+      // https://stackoverflow.com/questions/24077117/localstorage-in-win8-1-ie11-does-not-synchronize
+      if (isIE11OrLess()) {
+        setTimeout(() => emitEventsAndResetExpireEventTimeouts(), 1000);
```

Comment:
I'm concerned what may happen within that 1 second. Will we be emitting adding/removed events that could clobber more recent, authoritative values? Should we check that the current value in storage is the newToken?

Reply:
Good point, I am thinking how about attach a timestamp to the emitted events, if by any chance an old event comes to updateAuthState, just ignore it.

Reply:
^ smart catch